### PR TITLE
Set controller status before it is cheked on trajectory execution.

### PR DESCRIPTION
**Pull request by destogl**
### Description

This PR enalbes trajectory execution when using simulation.
The PR adds controllers' status before it is checked if all needed controller are activated.

This is neede to use MoveIt with Ignition Simulation (I will link UR example later today for this).
Though, the interesting part is, why trajectory execution is working without this changes when using "FakeSystem" or hardware.
It looks to me that the same checks should fail in any case.

I tested this with and without "moveit_manage_controllers" parameter set. And the changes are required in both cases.

I using "moveit_simple_controller_manager/MoveItSimpleControllerManager" in our setup. Is this maybe an issue?
Should another controller manager be used.

BTW, I see that MoveIs is not activelly checking which controllers are active and which not, although there is a management code for this implemented in `ensureActiveControllers` (line: 1718) of `trajectory_execution_manager.cpp`.

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by vatanaksoytezer** [comment]
@destogl clang-tidy seems unhappy of your changes, doo you mind re-basing and fixing clang-tidy suggestions?

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp [hunk 1]
```diff
@@ -567,6 +567,17 @@ void TrajectoryExecutionManager::reloadControllerInformation()
       known_controllers_[ci.name_] = ci;
     }
 
+    names.clear();
+    controller_manager_->getActiveControllers(names);
+    for (std::map<std::string, ControllerInformation>::iterator it = known_controllers_.begin();
```
Comment (henningkayser) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp:572]: I think with new code we should aim to use range-based loops or std algorithms. Also, is there a chance that `getActiveControllers()` returns names for which we don't have a known controller? If no, iterating names and accessing the map directly would make more sense to me

**Review by henningkayser — APPROVED** [review]

**Review by JafarAbdi — APPROVED** [review]

**Comment by destogl** [comment]
can we merge this? I think it has all approvals and it is up-to-date

**Comment by destogl** [comment]
@mergifyio backport galactic

**Pull request merged**